opsdroid/configuration/shema.yaml

```
@@ -0,0 +1,280 @@
+
+  ## Facebook (https://github.com/opsdroid/connector-facebook)
+  - name: str(required=False)
+    # required
+    verify-token: str()
+    page-access-token: str()
+    # optional
+    bot-name: str(required=False)
+
+  ## Github (https://github.com/opsdroid/connector-github)
+  - name: str(required=False)
+    # required
+    github-token: str()
+
+  ## Matrix (https://github.com/opsdroid/connector-matrix)
+  - name: str(required=False)
+    # Required
+    mxid: str()
+    password: str()
+    # Name of a single room to connect to
+    room: str()
+    # Alternatively, a dictionary of multiple rooms
+    # One of these should be named 'main'
+    rooms:
+      'main': str()
+      'other': str()
+    # Optional
+    homeserver: str(required=False)
+    nick: str(required=False)
+
+  ## Shell (https://github.com/opsdroid/connector-shell)
+  - name: str(required=False)
+    # optional
+    bot-name: str(required=False)
+
+  ## Slack (https://github.com/opsdroid/connector-slack)
+  - name: str(required=False)
+    # required
+    api-token: str()
+    # optional
+    bot-name: str(required=False)
+    default-room: str(required=False)
+    icon-emoji: str(required=False)
+    connect-timeout: int(required=False)
+
+  ## Telegram (https://github.com/opsdroid/connector-telegram)
+  - name: str() #telegram
+    # required
+    token: str() #"123456789:ABCDEFGHIJKLMNOPQRSTUVWXYZ-ZYXWVUT"  # Telegraf bot token
+    # optional
+    update_interval: num(required=False)     #0.5  # Interval between checking for messages
+    default_user: str(required=False) #user1  # Default user to send messages to (overrides default room in connector)
+    whitelisted_users: list() # List of users who can speak to the bot, if not set anyone can speak
+
+
+
+  ## Twitter (https://github.com/opsdroid/connector-twitter)
+  - name: str(required=False)
+    # required
+    consumer_key: str()
+    consumer_secret: str()
+    oauth_token: str()
+    oauth_token_secret: str()
+    # optional
+    enable_dms: bool(required=False)
+    enable_tweets: bool(required=False)
+
+  ## Websocket (https://github.com/opsdroid/connector-websocket)
+  - name: str(required=False)
+    # optional
+    bot-name: str(required=False)
+    max-connections: int(required=False)
+    connection-timeout: int(required=False)
+
+
+## Database modules (optional)
+
+  ## Mongo (https://github.com/opsdroid/database-mongo)
+databases: list(include('mongo', required=False),include('redis', required=False),include('sqlite', required=False))
+---
+mongo:
+  name: regex('^mongo$')
+  host: str(required=False)
+  port: str(required=False)
+  database: str(required=False)
+
+
+redis:
+  name: regex('^redis$')
+  host: str(required=False)
+  port: str(required=False)
+  database: str(required=False)
+  password: str(required=False)
+  reconnect: bool(required=False)
+
+
+sqlite:
+    name: regex('^sqlite$')
+    file: regex('(\\\\?([^\\/]*[\\/])*)([^\\/]+)$')
+    table: regex('(^[a-z_]+[0-9a-z]*$)')
```

```suggestion
    table: str(required=False)
```

Because https://stackoverflow.com/a/3694291/467145